ably/state.go

```diff
@@ -219,7 +219,7 @@ func (s *stateEmitter) set(state StateEnum, err error) error {
 	changed := s.current != state
 	s.current = state
 	s.err = stateError(state, err)
-	if changed {
+	if changed || err != nil {
```

Comment:
Why do we need to check `err!=nil`?  aren't we interested on state change only? We are already passing err on `s.err`